site/gatsby-site/src/pages/summaries/incidents.js

```diff
@@ -1,105 +1,97 @@
-import React, { Component } from 'react';
+import React from 'react';
 import Helmet from 'react-helmet';
 import { graphql } from 'gatsby';
-
 import Button from 'react-bootstrap/Button';
-
 import Layout from 'components/Layout';
 import Link from 'components/ui/Link';
 import { StyledHeading, StyledMainWrapper } from 'components/styles/Docs';
 
 const ReportList = ({ items }) => {
   return (
     <ul>
-      {items.map(({ node: { id, url, title } }) => (
-        <li key={id}>
-          <a href={url}>{title}</a>
+      {items.map((report) => (
+        <li key={report.report_number} data-cy={`report-${report.report_number}`}>
+          <a href={report.url}>{report.title}</a>
         </li>
       ))}
     </ul>
   );
 };
 
-const IncidentList = ({ group }) => {
+const IncidentList = ({ incidents }) => {
   return (
     <>
-      {group.map((value, idx) => (
-        <div key={`incident-${idx}`}>
+      {incidents.map((incident) => (
+        <div key={incident.incident_id} data-cy={`incident-${incident.incident_id}`}>
           <h2>
-            Incident {value['edges'][0]['node']['incident_id']}{' '}
-            <Button
-              variant="outline-primary"
-              href={'/cite/' + value['edges'][0]['node']['incident_id']}
-            >
+            Incident {incident.incident_id}{' '}
+            <Button variant="outline-primary" href={'/cite/' + incident.incident_id}>
               Citation
             </Button>
             <Button
               variant="outline-primary"
-              href={'/apps/discover?incident_id=' + value['edges'][0]['node']['incident_id']}
+              href={'/apps/discover?incident_id=' + incident.incident_id}
             >
               Discover
             </Button>
           </h2>
-          <ReportList items={value['edges']} />
+          <ReportList items={incident.reports} />
         </div>
       ))}
     </>
   );
 };
 
-export default class Incidents extends Component {
-  render() {
-    const { data } = this.props;
-
-    if (!data) {
-      return null;
-    }
-    const {
-      allMongodbAiidprodReports: { group },
-    } = data;
+export default function Incidents({ data, ...props }) {
+  const incidents = data.allMongodbAiidprodIncidents.nodes.map((incident) => {
+    const reports = incident.reports.map(
+      (report_number) =>
+        data.allMongodbAiidprodReports.nodes.find((r) => r.report_number == report_number) || {
+          report_number,
+          title: `Missing Report ${report_number}`,
+        } // there are missing reports, remove once that's fixed
```

Comment:
What is happening here?

Reply:
I forgot to flag this.

I'll have to look a little more into this, but there seem to be a couple of incidents that contain reports that do not exist:

- `report_number` is present in the `incidents.reports` array 
- there is no report in the `reports` collection with that `report_number.`

It is probably because of the inconsistency bug we had a while ago; it might be good to add a test for it next to the other integrity tests.